content-cache-backends-config/src/state.py

```diff
@@ -126,8 +130,96 @@ class Configuration(pydantic.BaseModel):
         pydantic.StringConstraints(min_length=1),
         pydantic.AfterValidator(_validate_path_value),
     ]
+    healthcheck_path: typing.Annotated[
+        str,
+        pydantic.StringConstraints(min_length=1),
+        pydantic.AfterValidator(_validate_path_value),
+    ]
+    healthcheck_interval: pydantic.PositiveInt
     proxy_cache_valid: tuple[str, ...]
 
+    @pydantic.field_validator("hostname")
+    @classmethod
+    def validate_hostname(cls, value: str) -> str:
+        """Validate the hostname.
+
+        Args:
+            value: The value to validate.
+
+        Raises:
+            ValueError: The validation failed.
+
+        Returns:
+            The value after validation.
+        """
+        if len(value) > 255:
+            raise ValueError("Hostname cannot be longer than 255")
+
+        valid_segment = re.compile(r"(?!-)[A-Z\d-]{1,63}(?<!-)$", re.IGNORECASE)
+        for segment in value.split("."):
+            if valid_segment.fullmatch(segment) is None:
+                raise ValueError(
+                    (
+                        "Each Hostname segment must be less than 64 in length, and consist of "
+                        "alphanumeric and hyphen"
+                    )
+                )
+
+        return value
+
+    @pydantic.field_validator("path")
+    @classmethod
+    def validate_path(cls, value: str) -> str:
+        """Validate the path.
+
+        Args:
+            value: The value to validate.
+
+        Returns:
+            The value after validation.
+        """
+        return _validate_path_value(value)
+
+    @pydantic.field_validator("fail_timeout")
+    @classmethod
+    def validate_fail_timeout(cls, value: str) -> str:
+        """Validate the fail_timeout.
+
+        Args:
+            value: The value to validate.
+
+        Returns:
+            The value after validation.
+        """
+        _check_nginx_time_str(value)
+        return value
+
+    @pydantic.field_validator("backends_path")
+    @classmethod
+    def validate_backends_path(cls, value: str) -> str:
+        """Validate the backends_path.
+
+        Args:
+            value: The value to validate.
+
+        Returns:
+            The value after validation.
+        """
+        return _validate_path_value(value)
+
+    @pydantic.field_validator("healthcheck_path")
+    @classmethod
+    def validate_healthcheck_path(cls, value: str) -> str:
+        """Validate the healthcheck_path.
+
+        Args:
+            value: The value to validate.
+
+        Returns:
+            The value after validation.
+        """
+        return _validate_path_value(value)
+
```

Comment:
I think line 141 to 222 should be removed.

The code was refactored to use `pydantic.AfterValidator`. E.g., for the `hostname`:
```
hostname: typing.Annotated[
        str,
        pydantic.StringConstraints(min_length=1),
        pydantic.AfterValidator(_validate_hostname_value),
    ]
```

I think due to merge conflict this part is being added back.